### PR TITLE
JiConOp2

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: modules/xmpp/Caps.js [hunk 1]
```diff
@@ -187,28 +187,35 @@ export default class Caps extends Listenable {
...
-                    .each(
-                        (_, el) => identities.add({
-                            type: el.getAttribute('type'),
-                            name: el.getAttribute('name'),
-                            category: el.getAttribute('category')
-                        }));
-                resolve({
-                    features,
-                    identities });
+                resolve(this.parseDiscoInfo(response));
```
Comment (bgrozev) [inline modules/xmpp/Caps.js:190]: I'm not sure I understand the js well, but does `parseDiscoInfo` need to be defined on the object? It looks like it only operates on its `node` parameter.

File: modules/xmpp/xmpp.js [hunk 2]
```diff
@@ -489,6 +489,20 @@ export default class XMPP extends Listenable {
 
         this.sendDiscoInfo = true;
 
+        if (this.connection._stropheConn && this.connection._stropheConn._addSysHandler) {
+            this.connection._stropheConn._addSysHandler(msg => {
+                this.sendDiscoInfo = false;
```
Comment (bgrozev) [inline modules/xmpp/xmpp.js:494]: Is this a one-time handler? If the XMPP server does not send the expected message, is this going to fire with the next regular `message` stanza we receive?

File: modules/xmpp/Caps.js (hunk 1 above)
Comment (damencho) [reply modules/xmpp/Caps.js:190]: It is defined in the object and same method is used in another place ... an option is to move it in util folder so can be used from both places ... but we will extract xmpp only and caps only functionality in another place ... so I don't see the point of that.

File: modules/xmpp/xmpp.js (hunk 2 above)
Comment (damencho) [reply modules/xmpp/xmpp.js:494]: this will fire anytime for messages with type conference-info.

File: modules/xmpp/xmpp.js [hunk 3]
```diff
@@ -498,6 +498,15 @@ export default class XMPP extends Listenable {
                 const { features, identities } = this.caps.parseDiscoInfo(msg);
 
                 this._processDiscoInfoIdentities(identities, features);
+
+                // check for shard name in identities
+                if (identities) {
```
Comment (bgrozev) [inline modules/xmpp/xmpp.js:503]: I think this test is redundant (even an empty Set evaluates to `true`), but I don't know if there's a way to express that in js.

Comment (damencho) [reply modules/xmpp/xmpp.js:503]: Ups, I will drop it.

File: modules/xmpp/xmpp.js (hunk 2 above)
Comment (bgrozev) [reply modules/xmpp/xmpp.js:494]: Ahh, I missed the other arguments. What makes a message have a type of "conference-info"? It contains a "<conference-info" element?

Nit: we haven't joined a conference yet, is for example "service-info" a more accurate name?

Comment (damencho) [reply modules/xmpp/xmpp.js:494]: https://github.com/jitsi/jitsi-meet/pull/9052/commits/1840d4d0bc4bd468c98ee9d7b52d41acba8cd334#diff-51cb31fcf1135ed84b83d4324d1086d258898f786d53cecf56c3d8ab85e07413R28
We set it, ok. Make sense, I will change it.

Comment (damencho) [reply modules/xmpp/xmpp.js:494]: Fixed.

File: modules/xmpp/Caps.js (hunk 1 above)
Comment (damencho) [reply modules/xmpp/Caps.js]: I fixed it with the latest commit.

File: modules/xmpp/xmpp.js (hunk 3 above)
Comment (damencho) [reply modules/xmpp/xmpp.js]: Done.

**Review by bgrozev — APPROVED** [review]

**Pull request merged**